### ISSUE TITLE
remove client side URL rewrites

### ISSUE THREAD

**Issue by wkeese** [lbod/dapi#53]
If I select 1.4 in the dropdown and bring up dijit/Calendar, then click the Permalink, the URL bar goes to https://dojotoolkit.org/api/?qs=1.4/dijit/Calendar.   Happens for 1.3 〜 1.7.

Two issues:
- Ideally it would be https://dojotoolkit.org/dapi/1.10/dijit/Calendar.html rather than using the URL query string (question mark syntax).
- The URL switched from referencing dapi to api.   I guess that won't be an issue when this is deployed for real, but probably indicates some underlying problem.

This problem is leftover effect from when we were serving pages statically and the rewrite was happening in client-side javascript.   By turning off javascript and clicking the Permalink URL (and then view source) I can see:

``` html
<script type="text/javascript">
  // If user loads this page directly, redirect to index page, with parameter to initially show tab for this module.
  // If google loads this page directly it will hopefully ignore this script and just index the documentation below.
  // Also, when this page is loaded via XHR this script block should be ignored.
  location = "/api/?qs=1.10/dijit/_OnDijitClickMixin" + location.hash;

</script>
```

I remember having a perl script that added those `<script>` blocks but now I can't find the check-in.  I don't see those script blocks [anymore] either in https://github.com/wkeese/dojo-api-data/tree/master/1.6/dijit/Calendar, so I'm confused.

There's also this code in app.js:

``` js
// rewite the url to the ?qs= parm, meaning all legacy permalinked docs are still loaded via XHR.
// the only other option is dynamically including the html in the template which wouldn't be good for performance
// TODO: the .html extension is an issue, here im stripping it (compared to legacyfilelocation), this is also mirrored in the client JS (and FWIW apache)
res.redirect(301, config.contextPath + '?qs=' + requestedVersion + '/' + modulefile.replace(/\./g, "/"));
```

A related issue is that after doing the above, the dropdown says 1.10 instead of 1.4, even though the tree is showing 1.4.   Selecting 1.10 in the dropdown list manually now has no effect.  (But I can get back to the 1.10 tree by selecting 1.9 first, then selecting 1.10).

**Comment by wkeese** [comment]
PS: It's possible I got confused about the `<script>` block from when it automatically switched the URL from https://dojotoolkit.org/dapi to https://dojotoolkit.org/api.  Need to doublecheck that.

**Comment by lbod** [comment]
Yes, that stuff is messy as hell and I've seen the versionSelector not working for legacy, I was sure it was at some point. Also remember that the permalinks for legacy are hardcoded in the HTML, as they were pre-generated i.e. 

```
   <div class="jsdoc-permalink" style="display:none;">/api/1.7/dojo</div>
```

It's also explained this is expected in https://github.com/dojo/dapi/wiki/Legacy-api-docs#caveat

About switching from /dapi to /api I'm also slightly concerned that dapi is being tested on "gnat' where I don't think the apache rules are actually in place, so I'm concerned we're not testing on 'staging' as expected in production.
Where are those apache rules again? Need to make sure they won't have any undesired effect when they're in use again.

Also, about the `location = ""` redirect. I remember we had a long discussion about that and I still don't really understand it's need now but... that markup should only be created by `node spider`.

**Comment by wkeese** [comment]
> Where are those apache rules again? Need to make sure they won't have any undesired effect when they're in use again.

There are some apache rules at https://github.com/wkeese/dojo-api-data/blob/1d802f0e51fec625702dd16bf9bdd1c3733b6091/.htaccess.
It looks like you don't need any of that because `legacyNSReplacers` in config.js is doing the same thing.

> Also, about the location = "" redirect. I remember we had a long discussion about that and I still don't really understand it's need now but... that markup should only be created by node spider.

Agreed.  OK, if it's only created by `node spider` then there's no issue.
